zarr/core.py

```diff
@@ -711,7 +719,7 @@ def _get_basic_selection_zd(self, selection, out=None, fields=None):
 
         except KeyError:
             # chunk not initialized
-            chunk = np.zeros((), dtype=self._dtype)
+            chunk = np.zeros_like(self._meta, shape=(), dtype=self._dtype)
```

Comment:
Beware, this may restrict you to a somewhat recent version of Numpy

Reply:
(was just writing that)

Reply:
You can always have a safe wrapper for this, like we do in dask: https://github.com/dask/dask/blob/master/dask/array/utils.py#L288-L334

Reply:
Yeah this is tricky if the storage layer is already producing an array of a different type, this will then cause a conversion. With CuPy for example, we know this is not an option ( https://github.com/cupy/cupy/issues/589 ). I'm not sure if we shouldn't just error out earlier.

Reply:
Not sure I understand, why won't CuPy work? CuPy supports the `*_like` functions in here since https://github.com/cupy/cupy/pull/2171.

Reply:
Meaning that when we can't use `*_like` functions we will create a NumPy array here, but we may be trying to store CuPy arrays into it. So this would fail because the CuPy arrays would be coerced to NumPy arrays. This is just an example.

Reply:
Yeah, I guess that goes back to the array creation issue that we're trying to address in NumPy. :)

Reply:
Yep :)